### PR TITLE
Added mining drill and diamond drill

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
## Short Description
Adds the Mining Drill and Diamond Drill. The diamond drill can not be upgraded to netherite via the Smithing Table.

## Additions/Changes/Removals
- Added Mining Drill class
- Added Diamond Drill class
- Added both items to Items
- Registered both items in ItemSetup

## Related Issues
Was on the LX to-do list

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

Here's a video: https://streamable.com/k3cnno

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java]: we have an events class

File: src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java]: this also has to go since we have an events class

File: src/main/java/dev/j3fftw/litexpansion/items/MiningDrill.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/MiningDrill.java]: also event class

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:176]: New line

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Why just overworld blocks?

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: das tru. what other blocks? end stone netherrack... basalt?

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: Everything a iron pick except ores could break for tier 1 and the highest tier everything a diamond pick could break?

Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: include post 1.14 blocks?

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: What do you mean

Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: basalt, blackstone

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: Is that even in minecraft if so yes include does but you need to check minecraft version

Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: or i could use collection and check if each material name contains the word but idk about the efficiency for that

not even sure about the efficiency for the current arraylist that im using, too many cant be good

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: formatting

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: || new line

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: should also break ores
and should be enchanted

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: spaces?

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: no the line is way to long

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: If it breaks ores, it will drop the ore block, unless I add special cases for every ore, or at this point I can change it to a pickaxe and fix all that

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: its okay for it dropping the ore block

File: src/main/java/dev/j3fftw/litexpansion/GlowEnchant.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/GlowEnchant.java]: Just the diamond

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:37]: *private final*

Also use base Types, so `List<Material>`

Comment (TheBusyBiscuit) [reply src/main/java/dev/j3fftw/litexpansion/Events.java:37]: Also you may probably rather wanna use a Set<Material> here since you don't really need order or double-entries, then I'd recommend an EnumSet.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/DiamondDrill.java]: https://wiki.industrial-craft.net/index.php?title=Diamond_Drill
Change recipe

File: src/main/java/dev/j3fftw/litexpansion/items/MiningDrill.java
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/MiningDrill.java]: https://wiki.industrial-craft.net/index.php?title=Mining_Drill
change recipe to the legacy recipe

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Comment by J3fftw1** [comment]
Add research

**Comment by J3fftw1** [comment]
wait for #80 to be merged
also update to it

**Comment by J3fftw1** [comment]
Do you mind updating it?

File: dependency-reduced-pom.xml
Comment (J3fftw1) [inline dependency-reduced-pom.xml:122]: what the hell is this

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
maybe add both drills in 1 file

File: src/main/java/dev/j3fftw/litexpansion/Events.java
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java:37]: cant we use a tag here for stones?

File: dependency-reduced-pom.xml
Comment (NCBPFluffyBear) [reply dependency-reduced-pom.xml:122]: what it appeared after i merged

Comment (J3fftw1) [reply dependency-reduced-pom.xml:122]: Not in my code but add it to gitignore

**Pull request merged**
